File: C#/Engine.Core/Maths/Limits.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Engine.Core.Maths
{
    public struct Limits
    {
        public float lowerLimit;
        public float upperLimit;
        /// <summary>
        /// The increment is the internal step width
        /// </summary>
        public float increment;

        public bool Contains(float f)
        {
            bool res = false;
            if (f >= lowerLimit && f <= upperLimit)
            {
                res = true;
                if (increment != 0)
                {
                    float fbyi = f / increment;
                    int x = Mathf.RoundToInt(fbyi);
                    float y = fbyi - x;
                    res = Mathf.Approximately(y, 0);
                }
            }
            return res;
        }
    }
}

File: Engine/src/Core/Maths/Mathf/Limits.cpp
#include "Limits.h"
#include "Mathf.h"

namespace Engine::Core::Maths
{
    Limits::Limits(float lowerLimit, float upperLimit, float increment) : Interval(IntervalType::Closed, lowerLimit, upperLimit), lowerLimit(lowerLimit), upperLimit(upperLimit)
    {
        if (increment != 0.0f)
        {
            this->increment = increment;
        }

        if (lowerLimit > upperLimit)
        {
            throw std::invalid_argument("Lower limit cannot be greater than upper limit");
        }
    }

    bool Limits::Contains(const float &f) const
    {
        bool res = false;
        if (f >= Limits::lowerLimit && f <= Limits::upperLimit)
        {
            res = true;
            if (Limits::increment != 0)
            {
                float fbyi = f / Limits::increment;
                int x = Mathf::RoundToInt(fbyi);
                float y = fbyi - x;
                res = Mathf::Approximately(y, 0);
            }
        }
        return res;
    }
}